CONTRIBUTING.md

```diff
@@ -0,0 +1,111 @@
+* If the `<plugin>` is in the groupId `org.apache.maven.plugins` you shall omit the `<groupId>`.
+* All `<plugin>` entries shall have an explicit version defined unless inherited from the parent.
+
+## Java code style
+
+### Imports
+
+* For code in `src/main`:
+    - `*` imports are banned. 
+    - `static` imports are preferred until not mislead.
+* For code in `src/test`:
+    - `*` imports of anything other than JUnit classes and Hamcrest matchers are banned.
+
+### Annotation placement
+
+* Annotations on classes, interfaces, annotations, enums, methods, fields and local variables shall be on the lines immediately preceding the line where modifier(s) (e.g. `public` / `protected` / `private` / `final`, etc) would be appropriate.
+* Annotations on method arguments shall, to the largest extent possible, be on the same line as the method argument (and, if present, before the `final` modifier)
+
+### Javadoc
+
+* Each class shall have a Javadoc comment.
+* Unless the method is `private`, it shall have a Javadoc comment.
+* Getters and Setters shall have a Javadoc comment. The following is prefered
+    ```
+    /**
+     * The count of widgets
+     */
+    private int widgetCount;
+    
+    /**
+     * Returns the count of widgets.
+     *
+     * @return the count of widgets. 
+     */
+    public int getWidgetCount() {
+        return widgetCount;
+    }
+    
+    /**
+     * Sets the count of widgets.
+     *
+     * @param widgetCount the count of widgets.
+     */
+    public void setWidgetCount(int widgetCount) {
+        this.widgetCount = widgetCount;
+    }
+    ```
+* When adding a new class / interface / etc, it shall have a `@since` doc comment. The version shall be `FIXME` (or `TODO`) to indicate that the person merging the change should replace the `FIXME` with the next release version number. The fields and methods within a class/interface (but not nested classes) will be assumed to have the `@since` annotation of their class/interface unless a different `@since` annotation is present.
+
+### IDE Configuration
+
+* Eclipse, by and large the IDE defaults are acceptable with the following changes:
+    - Tab policy to `Spaces only`
+    - Indent statements within `switch` body
+    - Maximum line width `120`
+    - Line wrapping, ensure all to `wrap where necessary`
+    - Organize imports alphabetically, no grouping
+* NetBeans, by and large the IDE defaults are acceptable with the following changes:
+    - Tabs and Indents
+        + Change Right Margin to `120`
+        + Indent case statements in switch
+    - Wrapping
+        + Change all the `Never` values to `If Long`
+        + Select the checkbox for Wrap After Assignement Operators
+* IntelliJ, by and large the IDE defaults are acceptable with the following changes:
+    - Wrapping and Braces
+        + Change `Do not wrap` to `Wrap if long`
+        + Change `Do not force` to `Always`
+    - Javadoc
+        + Disable generating `<p/>` on empty lines
```

Comment:
it's ugly. We not going to generate javadoc site, so it should be readable as is. `<p/>` is not for reading it as is